File: java/com/epicdragonworld/gameserver/data/ItemData.java
package com.epicdragonworld.gameserver.data;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.util.HashMap;
import java.util.Map;
import java.util.logging.Logger;

import com.epicdragonworld.gameserver.enums.ItemType;
import com.epicdragonworld.gameserver.managers.DatabaseManager;
import com.epicdragonworld.gameserver.model.holders.ItemHolder;
import com.epicdragonworld.gameserver.model.holders.SkillHolder;

/**
 * @author Pantelis Andrianakis
 */
public class ItemData
{
	private static final Logger LOGGER = Logger.getLogger(ItemData.class.getName());
	
	private static final String RESTORE_ITEMS = "SELECT * FROM items";
	
	private static final Map<Integer, ItemHolder> _items = new HashMap<>();
	
	public static void init()
	{
		_items.clear();
		
		try (Connection con = DatabaseManager.getConnection();
			PreparedStatement ps = con.prepareStatement(RESTORE_ITEMS))
		{
			try (ResultSet rset = ps.executeQuery())
			{
				while (rset.next())
				{
					final int itemId = rset.getInt("item_id");
					final int skillId = rset.getInt("skill_id");
					final int skillLevel = rset.getInt("skill_level");
					final SkillHolder skillHolder = SkillData.getSkillHolder(skillId, skillLevel);
					if ((skillId > 0) && (skillHolder == null))
					{
						LOGGER.warning("ItemData: Could not find skill with id " + skillId + " and level " + skillLevel + " for item " + itemId + ".");
					}
					else
					{
						_items.put(itemId, new ItemHolder(itemId, rset.getInt("slot_id"), Enum.valueOf(ItemType.class, rset.getString("type")), rset.getBoolean("stackable"), rset.getBoolean("tradable"), rset.getLong("price"), rset.getInt("stamina"), rset.getInt("strength"), rset.getInt("dexterity"), rset.getInt("intelect"), skillHolder));
					}
				}
			}
		}
		catch (Exception e)
		{
			LOGGER.warning(e.getMessage());
		}
		
		LOGGER.info("ItemData: Loaded " + _items.size() + " items.");
	}
	
	public static ItemHolder getItemHolder(int itemId)
	{
		return _items.get(itemId);
	}
}

File: EpicDragonWorldServer/Data/ItemData.cs
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;

/**
 * Author: Pantelis Andrianakis
 * Date: May 5th 2019
 */
public class ItemData
{
    private static protected string RESTORE_ITEMS = "SELECT * FROM items";
    private static protected Dictionary<int, ItemHolder> ITEMS = new Dictionary<int, ItemHolder>();

    public static void Load()
    {
        Util.PrintSection("Items");

        ITEMS.Clear();
        try
        {
            MySqlConnection con = DatabaseManager.GetConnection();
            MySqlCommand cmd = new MySqlCommand(RESTORE_ITEMS, con);
            MySqlDataReader reader = cmd.ExecuteReader();
            while (reader.Read())
            {
                int itemId = reader.GetInt32("item_id");
                int skillId = reader.GetInt32("skill_id");
                int skillLevel = reader.GetInt32("skill_level");
                SkillHolder skillHolder = SkillData.GetSkillHolder(skillId, skillLevel);
                if ((skillId > 0) && (skillHolder == null))
                {
                    LogManager.Log("ItemData: Could not find skill with id " + skillId + " and level " + skillLevel + " for item " + itemId + ".");
                }
                else
                {
                    ITEMS.Add(itemId, new ItemHolder(itemId, (ItemSlot)Enum.Parse(typeof(ItemSlot), reader.GetString("slot")), (ItemType)Enum.Parse(typeof(ItemType), reader.GetString("type")), reader.GetBoolean("stackable"), reader.GetBoolean("tradable"), reader.GetInt32("stamina"), reader.GetInt32("strength"), reader.GetInt32("dexterity"), reader.GetInt32("intelect"), skillHolder));
                }
            }
            con.Close();
        }
        catch (Exception e)
        {
            LogManager.Log(e.ToString());
        }

        LogManager.Log("ItemData: Loaded " + ITEMS.Count + " items.");
    }

    public static ItemHolder GetItemHolder(int itemId)
    {
        return ITEMS[itemId];
    }
}
